### PR TITLE
Implement scrollbar to list

### LINKED ISSUE #609: add slidder to pages that scroll.
Use the scroll bar widget on pages that have to scroll like the transaction page, wallet log page etc.
This is to enable us have a drag-able slider on the right hand side of the screen.
The slider should only be visible when the item on the page is more than the page height, and when the pointer gets to the right hand side of the screen. 

Kindly look at this PR for insight https://github.com/planetdecred/godcr/pull/253

### PR DESCRIPTION
**Pull request by JustinBeBoy**
Resolve #609
- Upgrade gioui new version
- Implement scrollbar for list
- Update current component for gioui new version
- Custom Icon

scroll1

scroll2

scroll3

### CODE REVIEW

**Comment by Sirmorrison** [comment]
I dont think you will need those custom widgets you are adding since the recent updates to the gioui package has a scrollbar widget attached.
https://github.com/gioui/gio/blob/main/widget/list.go
Correct me if i am wrong but i believe this is what we want.

**Comment by Sirmorrison** [comment]
Close #665

**Comment by Sirmorrison** [comment]
Kindly rebase this Pr as the gioui library has been updated to a more recent version.

also, from the first screenshot your dropped, the scrollbar widget does not align with the page content area.. that the the area that is to scroll.
image

The second screenshot aligns well to the scroll area, however, it is too close to the content card.. there should be some padding in-between.
image

The 3rd screenshot has it scroll bar inside the content area. It should be outside like the others.
image

File: ui/decredmaterial/icon.go [hunk 1]
```diff
@@ -3,20 +3,32 @@
...
+	"image/color"
 	_ "image/png" //makes png images a decodable format
 
 	"gioui.org/widget"
 )
 
 type Icon struct {
 	*widget.Icon
+	Color color.NRGBA
+	Size  int
```
Comment (Sirmorrison) [inline ui/decredmaterial/icon.go]: This is redundant. Kindly remove.

File: ui/decredmaterial/icon.go [hunk 2]
```diff
@@ -3,20 +3,32 @@
...
-	if err != nil {
-		return nil, err
+func NewIcon(icon *widget.Icon) *Icon {
+	return &Icon{
+		Icon: icon,
 	}
-	return &Icon{icon}, nil
+}
+
+func (icon *Icon) Layout(gtx C) D {
```
Comment (Sirmorrison) [inline ui/decredmaterial/icon.go]:
```suggestion
func (icon *Icon) Layout(gtx C, iconSize unit.Value) D {
```

File: ui/decredmaterial/icon.go [hunk 3]
```diff
@@ -3,20 +3,32 @@
...
+	cl := color.NRGBA{A: 0xff}
+	if icon.Color != (color.NRGBA{}) {
+		cl = icon.Color
+	}
+	if icon.Size != 0 {
+		gtx.Constraints.Min.X = icon.Size
+	}
```
Comment (Sirmorrison) [inline ui/decredmaterial/icon.go]:
```suggestion
	gtx.Constraints.Min.X = gtx.Px(iconSize)
```

File: ui/modal/info_modal.go [hunk 4]
```diff
@@ -189,8 +187,9 @@ func (in *InfoModal) Layout(gtx layout.Context) D {
 
 		return layout.Inset{Top: values.MarginPadding10, Bottom: values.MarginPadding20}.Layout(gtx, func(gtx C) D {
 			return layout.Center.Layout(gtx, func(gtx C) D {
-				gtx.Constraints.Min.X = gtx.Px(values.MarginPadding50)
-				return in.dialogIcon.Layout(gtx, in.Theme.Color.DeepBlue)
+				in.dialogIcon.Color = in.Theme.Color.DeepBlue
+				in.dialogIcon.Size = 50
+				return in.dialogIcon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/modal/info_modal.go]:
```suggestion
				return in.dialogIcon.Layout(gtx, values.MarginPadding50)
```

File: ui/modal/info_modal_layouts.go [hunk 5]
```diff
@@ -51,9 +51,10 @@ func privacyInfo(th *decredmaterial.Theme) []layout.Widget {
 		func(gtx C) D {
 			return layout.Flex{Alignment: layout.Baseline}.Layout(gtx,
 				layout.Rigid(func(gtx C) D {
-					ic := decredmaterial.MustIcon(widget.NewIcon(icons.ImageLens))
-					gtx.Constraints.Min.X = gtx.Px(values.MarginPadding8)
-					return ic.Layout(gtx, th.Color.Gray)
+					ic := decredmaterial.NewIcon(decredmaterial.MustIcon(widget.NewIcon(icons.ImageLens)))
+					ic.Color = th.Color.Gray
+					ic.Size = 8
+					return ic.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/modal/info_modal_layouts.go]:
```suggestion
					return ic.Layout(gtx, values.MarginPadding8)
```

File: ui/modal/info_modal_layouts.go [hunk 6]
```diff
@@ -70,9 +71,10 @@ func privacyInfo(th *decredmaterial.Theme) []layout.Widget {
...
```
Comment (Sirmorrison) [inline ui/modal/info_modal_layouts.go]:
```suggestion
					return ic.Layout(gtx, values.MarginPadding8)
```

File: ui/modal/text_input_modal.go [hunk 7]
```diff
@@ -139,11 +139,12 @@ func (tm *TextInputModal) Layout(gtx layout.Context) D {
...
 				layout.Rigid(func(gtx C) D {
-					img := tm.Icons.ActionInfo
+					img := decredmaterial.NewIcon(tm.Icons.ActionInfo)
+					img.Color = tm.Theme.Color.Gray3
 					inset := layout.Inset{Right: values.MarginPadding4}
 					return inset.Layout(gtx, func(gtx C) D {
-						gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-						return img.Layout(gtx, tm.Theme.Color.Gray3)
+						img.Size = 20
+						return img.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/modal/text_input_modal.go]:
```suggestion
						return img.Layout(gtx, values.MarginPadding20)
```

File: ui/page/about_page.go [hunk 8]
```diff
@@ -121,8 +119,8 @@ func (pg *AboutPage) layoutRows(gtx layout.Context) layout.Dimensions {
 					layout.Flexed(1, func(gtx C) D {
 						return layout.E.Layout(gtx, func(gtx C) D {
 							return in.Layout(gtx, func(gtx C) D {
-								gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-								return pg.chevronRightIcon.Layout(gtx, pg.Theme.Color.Gray)
+								pg.chevronRightIcon.Size = 20
+								return pg.chevronRightIcon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/about_page.go]:
```suggestion
								return pg.chevronRightIcon.Layout(gtx, values.MarginPadding20)
```

File: ui/page/components/account_selector.go [hunk 9]
```diff
@@ -167,8 +167,9 @@ func (as *AccountSelector) Layout(gtx layout.Context) layout.Dimensions {
 							Left: values.MarginPadding15,
 						}
 						return inset.Layout(gtx, func(gtx C) D {
-							gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-							return as.Icons.DropDownIcon.Layout(gtx, as.Theme.Color.Gray3)
+							ic := decredmaterial.NewIcon(as.Icons.DropDownIcon)
+							ic.Size = 20
+							return ic.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/components/account_selector.go]:
```suggestion
							return ic.Layout(gtx, values.MarginPadding20)
```

File: ui/page/components/account_selector.go [hunk 10]
```diff
@@ -434,8 +435,9 @@ func (asm *AccountSelectorModal) walletAccountLayout(gtx layout.Context, account
 			sections := func(gtx layout.Context) layout.Dimensions {
 				return layout.E.Layout(gtx, func(gtx C) D {
 					return inset.Layout(gtx, func(gtx C) D {
-						gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-						return asm.Icons.NavigationCheck.Layout(gtx, asm.Theme.Color.Gray3)
+						ic := decredmaterial.NewIcon(asm.Icons.NavigationCheck)
+						ic.Size = 20
+						return ic.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/components/account_selector.go]:
```suggestion
						return ic.Layout(gtx,values.MarginPadding20)
```

File: ui/page/debug_page.go [hunk 11]
```diff
@@ -79,8 +78,9 @@ func (pg *DebugPage) debugItem(gtx C, i int) D {
 		layout.Flexed(1, func(gtx C) D {
 			return layout.E.Layout(gtx, func(gtx C) D {
 				return layout.UniformInset(values.MarginPadding15).Layout(gtx, func(gtx C) D {
-					gtx.Constraints.Min.X = gtx.Px(values.MarginPadding22)
-					return pg.Icons.ChevronRight.Layout(gtx, pg.Theme.Color.Gray)
+					ic := decredmaterial.NewIcon(pg.Icons.ChevronRight)
+					ic.Size = 22
+					return ic.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/debug_page.go]:
```suggestion
					return ic.Layout(gtx, values.MarginPadding22)
```

File: ui/page/overview_page.go [hunk 12]
```diff
@@ -384,9 +387,10 @@ func (pg *OverviewPage) blockInfoRow(gtx layout.Context) layout.Dimensions {
 			}.Layout(gtx, pg.Theme.Body1(fmt.Sprintf("%d", pg.bestBlock.Height)).Layout)
 		}),
 		layout.Rigid(func(gtx C) D {
+			pg.walletStatusIcon.Color = pg.Theme.Color.Gray
+			pg.walletStatusIcon.Size = 5
```
Comment (Sirmorrison) [inline ui/page/overview_page.go]:
```suggestion
```

File: ui/page/overview_page.go [hunk 13]
```diff
@@ -384,9 +387,10 @@ func (pg *OverviewPage) blockInfoRow(gtx layout.Context) layout.Dimensions {
...
 			return layout.Inset{Right: values.MarginPadding5, Top: values.MarginPadding8}.Layout(gtx, func(gtx C) D {
-				gtx.Constraints.Min.X = gtx.Px(values.MarginPadding5)
-				return pg.walletStatusIcon.Layout(gtx, pg.Theme.Color.Gray)
+				return pg.walletStatusIcon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/overview_page.go]:
```suggestion
				return pg.walletStatusIcon.Layout(gtx, values.MarginPadding5)
```

File: ui/page/overview_page.go [hunk 14]
```diff
@@ -416,18 +420,19 @@ func (pg *OverviewPage) syncBoxTitleRow(gtx layout.Context) layout.Dimensions {
 	title := pg.Theme.Body2(values.String(values.StrWalletStatus))
 	title.Color = pg.Theme.Color.Gray3
 	statusLabel := pg.Theme.Body1(values.String(values.StrOffline))
-	clr := pg.Theme.Color.Danger
+	pg.walletStatusIcon.Color = pg.Theme.Color.Danger
+	pg.walletStatusIcon.Size = 14
 	if pg.isConnnected {
 		statusLabel.Text = values.String(values.StrOnline)
-		clr = pg.Theme.Color.Success
+		//clr = pg.Theme.Color.Success
```
Comment (Sirmorrison) [inline ui/page/overview_page.go]: Remove if no longer needed.

File: ui/page/overview_page.go [hunk 15]
```diff
@@ -416,18 +420,19 @@ func (pg *OverviewPage) syncBoxTitleRow(gtx layout.Context) layout.Dimensions {
 	title := pg.Theme.Body2(values.String(values.StrWalletStatus))
 	title.Color = pg.Theme.Color.Gray3
 	statusLabel := pg.Theme.Body1(values.String(values.StrOffline))
-	clr := pg.Theme.Color.Danger
+	pg.walletStatusIcon.Color = pg.Theme.Color.Danger
+	pg.walletStatusIcon.Size = 14
```
Comment (Sirmorrison) [inline ui/page/overview_page.go]:
```suggestion
```

File: ui/page/overview_page.go [hunk 16]
```diff
@@ -416,18 +420,19 @@ func (pg *OverviewPage) syncBoxTitleRow(gtx layout.Context) layout.Dimensions {
...
+		pg.walletStatusIcon.Color = pg.Theme.Color.Success
 	}
 
 	syncStatus := func(gtx layout.Context) layout.Dimensions {
 		return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
 			layout.Rigid(func(gtx C) D {
 				return layout.Inset{Top: values.MarginPadding4, Right: values.MarginPadding4}.Layout(gtx, func(gtx C) D {
-					gtx.Constraints.Min.X = gtx.Px(values.MarginPadding14)
-					return pg.walletStatusIcon.Layout(gtx, clr)
+					return pg.walletStatusIcon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/overview_page.go]:
```suggestion
					return pg.walletStatusIcon.Layout(gtx, values.MarginPadding14)
```

File: ui/page/overview_page.go [hunk 17]
```diff
@@ -468,8 +473,9 @@ func (pg *OverviewPage) syncStatusTextRow(gtx layout.Context, inset layout.Inset
 						}
 
 						return layout.Inset{Right: values.MarginPadding4}.Layout(gtx, func(gtx C) D {
-							gtx.Constraints.Min.X = gtx.Px(values.MarginPadding16)
-							return pg.cachedIcon.Layout(gtx, pg.Theme.Color.Gray)
+							pg.cachedIcon.Color = pg.Theme.Color.Gray
+							pg.cachedIcon.Size = 16
+							return pg.cachedIcon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/overview_page.go]:
```suggestion
							return pg.cachedIcon.Layout(gtx, values.MarginPadding16)
```

File: ui/page/overview_page.go [hunk 18]
```diff
@@ -491,18 +497,18 @@ func (pg *OverviewPage) syncStatusTextRow(gtx layout.Context, inset layout.Inset
...
 	}
 	i := layout.Inset{Right: values.MarginPadding16, Top: values.MarginPadding9}
 	if pg.walletSyncing {
 		return i.Layout(gtx, pg.syncingIcon.Layout24dp)
 	}
 	return i.Layout(gtx, func(gtx C) D {
-		gtx.Constraints.Min.X = gtx.Px(values.MarginPadding24)
-		return syncStatusIcon.Layout(gtx, col)
+		syncStatusIcon.Size = 24
+		return syncStatusIcon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/overview_page.go]:
```suggestion
		return syncStatusIcon.Layout(gtx, values.MarginPadding24)
```

File: ui/page/proposal/proposal_details_page.go [hunk 19]
```diff
@@ -59,12 +58,13 @@ func newProposalDetailsPage(l *load.Load, proposal *dcrlibwallet.Proposal) *prop
 		descriptionList:    &layout.List{Axis: layout.Vertical},
 		redirectIcon:       l.Icons.RedirectIcon,
 		downloadIcon:       l.Icons.DownloadIcon,
-		proposalItems:      make(map[string]proposalItemWidgets),
-		rejectedIcon:       l.Icons.NavigationCancel,
-		successIcon:        l.Icons.ActionCheckCircle,
-		timerIcon:          l.Icons.TimerIcon,
-		viewInPoliteiaBtn:  l.Theme.NewClickable(true),
-		voteBar:            NewVoteBar(l),
+		//voteBar:            l.Theme.VoteBar(decredmaterial.NewIcon(l.Icons.ActionInfo), decredmaterial.NewIcon(l.Icons.ImageBrightness1)),
```
Comment (Sirmorrison) [inline ui/page/proposal/proposal_details_page.go]: Remove if no longer needed.

File: ui/page/proposal/proposal_details_page.go [hunk 20]
```diff
@@ -190,14 +190,15 @@ func (pg *proposalDetails) layoutInDiscussionState(gtx C) D {
...
+				icon.Color = pg.Theme.Color.Primary
 				return layout.Inset{
 					Left:   values.MarginPaddingMinus2,
 					Right:  values.MarginPaddingMinus2,
 					Bottom: values.MarginPaddingMinus2,
 				}.Layout(gtx, func(gtx C) D {
-					gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-					return icon.Layout(gtx, pg.Theme.Color.Primary)
+					icon.Size = 24
+					return icon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/proposal/proposal_details_page.go]: Same as above

File: ui/page/proposal/proposal_details_page.go [hunk 21]
```diff
@@ -267,8 +266,8 @@ func (pg *proposalDetails) layoutNormalTitle(gtx C) D {
 					if icon == nil {
 						return D{}
 					}
-					gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-					return icon.Layout(gtx, iconColor)
+					icon.Size = 20
+					return icon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/proposal/proposal_details_page.go]: Same as above

File: ui/page/proposal/proposals_page.go [hunk 22]
```diff
@@ -493,8 +501,8 @@ func (pg *ProposalsPage) layoutAuthorAndDate(gtx C, item proposalItem) D {
 									}
 									rect.Max.Y = 20
 									pg.layoutInfoTooltip(gtx, rect, item)
-									gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-									return pg.infoIcon.Layout(gtx, pg.Theme.Color.Gray)
+									pg.infoIcon.Size = 20
+									return pg.infoIcon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/proposal/proposals_page.go]: Same as above

File: ui/page/proposal/proposals_page.go [hunk 23]
```diff
@@ -551,8 +559,8 @@ func (pg *ProposalsPage) layoutProposalVoteBar(gtx C, item proposalItem) D {
 func (pg *ProposalsPage) layoutIsSyncedSection(gtx C) D {
 	return layout.Flex{}.Layout(gtx,
 		layout.Rigid(func(gtx C) D {
-			gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-			return pg.updatedIcon.Layout(gtx, pg.Theme.Color.Success)
+			pg.updatedIcon.Size = 20
+			return pg.updatedIcon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/proposal/proposals_page.go]: Same as above

File: ui/page/proposal/votebar_widget.go [hunk 24]
```diff
@@ -232,8 +231,8 @@ func (v *VoteBar) layoutIconAndText(gtx C, lbl decredmaterial.Label, count float
 		return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
 			layout.Rigid(func(gtx C) D {
 				return layout.Inset{Right: values.MarginPadding5, Top: values.MarginPadding5}.Layout(gtx, func(gtx C) D {
-					gtx.Constraints.Min.X = gtx.Px(values.MarginPadding10)
-					return v.legendIcon.Layout(gtx, clr)
+					v.legendIcon.Size = 10
+					return v.legendIcon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/proposal/votebar_widget.go]: Same as above

File: ui/page/proposal/votebar_widget.go [hunk 25]
```diff
@@ -271,8 +270,8 @@ func (v *VoteBar) layoutInfo(gtx C) D {
 			rect.Max.Y = 20
 			v.layoutInfoTooltip(gtx, rect)
 			return layout.Inset{Left: values.MarginPadding5}.Layout(gtx, func(gtx C) D {
-				gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-				return v.infoIcon.Layout(gtx, v.Theme.Color.Gray)
+				v.infoIcon.Size = 20
+				return v.infoIcon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/proposal/votebar_widget.go]: Same as above

File: ui/page/proposal/wallet_selector.go [hunk 26]
```diff
@@ -142,8 +142,9 @@ func (as *WalletSelector) Layout(gtx layout.Context) layout.Dimensions {
 										Left: values.MarginPadding15,
 									}
 									return inset.Layout(gtx, func(gtx C) D {
-										gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-										return as.Icons.DropDownIcon.Layout(gtx, as.Theme.Color.Gray3)
+										ic := decredmaterial.NewIcon(as.Icons.DropDownIcon)
+										ic.Size = 20
+										return ic.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/proposal/wallet_selector.go]: Same as above

File: ui/page/proposal/wallet_selector.go [hunk 27]
```diff
@@ -319,8 +320,9 @@ func (asm *WalletSelectorModal) walletAccountLayout(gtx layout.Context, wallet *
 						sections := func(gtx layout.Context) layout.Dimensions {
 							return layout.E.Layout(gtx, func(gtx C) D {
 								return inset.Layout(gtx, func(gtx C) D {
-									gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-									return asm.Icons.NavigationCheck.Layout(gtx, asm.Theme.Color.Gray3)
+									ic := decredmaterial.NewIcon(asm.Icons.NavigationCheck)
+									ic.Size = 20
+									return ic.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/proposal/wallet_selector.go]: Same as above

File: ui/page/seedbackup/backup_success.go [hunk 28]
```diff
@@ -60,9 +60,9 @@ func (pg *BackupSuccessPage) Layout(gtx C) D {
 					Direction:   layout.Center,
 				}.Layout(gtx,
 					layout.Rigid(func(gtx C) D {
-						successIcon := pg.Icons.ActionCheckCircle
-						gtx.Constraints.Min.X = gtx.Px(values.MarginPadding24)
-						return successIcon.Layout(gtx, pg.Theme.Color.Success)
+						successIcon := decredmaterial.NewIcon(pg.Icons.ActionCheckCircle)
+						successIcon.Size = 64
+						return successIcon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/seedbackup/backup_success.go]: Same as above

File: ui/page/send/send_confirm_modal.go [hunk 29]
```diff
@@ -155,10 +155,11 @@ func (scm *sendConfirmModal) Layout(gtx layout.Context) D {
...
 					return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
 						layout.Rigid(func(gtx C) D {
-							icon := scm.Icons.NavigationArrowForward
+							icon := decredmaterial.NewIcon(scm.Icons.NavigationArrowForward)
+							icon.Color = scm.Theme.Color.Gray3
 							return layout.Inset{Right: values.MarginPadding8}.Layout(gtx, func(gtx C) D {
-								gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-								return icon.Layout(gtx, scm.Theme.Color.Gray3)
+								icon.Size = 15
+								return icon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/send/send_confirm_modal.go]: Same as above

File: ui/page/send/send_confirm_modal.go [hunk 30]
```diff
@@ -230,10 +231,11 @@ func (scm *sendConfirmModal) Layout(gtx layout.Context) D {
...
 			return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
 				layout.Rigid(func(gtx C) D {
-					icon := scm.Icons.ActionInfo
+					icon := decredmaterial.NewIcon(scm.Icons.ActionInfo)
+					icon.Color = scm.Theme.Color.Gray
 					return layout.Inset{Right: values.MarginPadding10}.Layout(gtx, func(gtx C) D {
-						gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-						return icon.Layout(gtx, scm.Theme.Color.Gray)
+						icon.Size = 20
+						return icon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/send/send_confirm_modal.go]: Same as above

File: ui/page/settings_page.go [hunk 31]
```diff
@@ -347,8 +346,8 @@ func (pg *SettingsPage) clickableRow(gtx layout.Context, row row) layout.Dimensi
 				return layout.Flex{}.Layout(gtx,
 					layout.Rigid(row.label.Layout),
 					layout.Rigid(func(gtx C) D {
-						gtx.Constraints.Min.X = gtx.Px(values.MarginPadding22)
-						return row.icon.Layout(gtx, pg.Theme.Color.LightGray)
+						row.icon.Size = 22
+						return row.icon.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/settings_page.go]: Same as above

File: ui/page/tickets/utils.go [hunk 32]
```diff
@@ -470,8 +470,9 @@ func ticketCard(gtx layout.Context, l *load.Load, tx *transactionItem, showWalle
 										Left:  values.MarginPadding4,
 										Right: values.MarginPadding4,
 									}.Layout(gtx, func(gtx C) D {
-										gtx.Constraints.Min.X = gtx.Px(values.MarginPadding5)
-										return l.Icons.ImageBrightness1.Layout(gtx, l.Theme.Color.Gray2)
+										ic := decredmaterial.NewIcon(l.Icons.ImageBrightness1)
+										ic.Size = 5
+										return ic.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/tickets/utils.go]: Same as above

File: ui/page/tickets/vsp_selector.go [hunk 33]
```diff
@@ -107,8 +107,9 @@ func (v *vspSelector) Layout(gtx layout.Context) layout.Dimensions {
 										Left: values.MarginPadding15,
 									}
 									return inset.Layout(gtx, func(gtx C) D {
-										gtx.Constraints.Min.X = gtx.Px(values.MarginPadding20)
-										return v.Icons.DropDownIcon.Layout(gtx, v.Theme.Color.Gray3)
+										ic := decredmaterial.NewIcon(v.Icons.DropDownIcon)
+										ic.Size = 20
+										return ic.Layout(gtx)
```
Comment (Sirmorrison) [inline ui/page/tickets/vsp_selector.go]: Same as above

**Review by Sirmorrison — CHANGES_REQUESTED** [review]
Kindly pay attention to reviews on the icons size parameter.

File: ui/decredmaterial/clickable_list.go [hunk 34]
```diff
@@ -15,15 +18,25 @@ type ClickableList struct {
...
 }
 
 func (t *Theme) NewClickableList(axis layout.Axis) *ClickableList {
-	return &ClickableList{
+	click := &ClickableList{
 		theme:          t,
 		List:           layout.List{Axis: axis},
 		ClickHighlight: t.Color.SurfaceHighlight,
 		selectedItem:   -1,
+		HideScroll:     false,
```
Comment (Sirmorrison) [inline ui/decredmaterial/clickable_list.go]: Rename to `IsHiddenScrollBer`

File: ui/page/proposal/proposals_page.go [hunk 35]
```diff
@@ -229,11 +230,18 @@ func (pg *ProposalsPage) OnClose() {
 
 func (pg *ProposalsPage) initLayoutWidgets() {
 	pg.categoryList = pg.Theme.NewClickableList(layout.Horizontal)
+	pg.categoryList.HideScroll = true
```
Comment (Sirmorrison) [inline ui/page/proposal/proposals_page.go]:
```suggestion
	pg.categoryList.IsHiddenScrollbar = true
```

File: ui/page/wallet_page.go [hunk 36]
```diff
@@ -357,11 +357,17 @@ func (pg *WalletPage) Layout(gtx layout.Context) layout.Dimensions {
...
+			//	return pg.layoutAddWalletSection(gtx)
+			//}),
 		)
 	}
+	layout.Stack{Alignment: layout.SE}.Layout(gtx,
+		layout.Stacked(func(gtx C) D {
+			return pg.layoutAddWalletSection(gtx)
+		}),
+	)
+	//pg.layoutAddWalletSection(gtx)
```
Comment (Sirmorrison) [inline ui/page/wallet_page.go]: If not needed, kindly remove.

File: ui/page/wallet_page.go [hunk 37]
```diff
@@ -357,11 +357,17 @@ func (pg *WalletPage) Layout(gtx layout.Context) layout.Dimensions {
 					return dims
 				})
 			}),
-			layout.Stacked(func(gtx C) D {
-				return pg.layoutAddWalletSection(gtx)
-			}),
+			//layout.Stacked(func(gtx C) D {
+			//	return pg.layoutAddWalletSection(gtx)
+			//}),
```
Comment (Sirmorrison) [inline ui/page/wallet_page.go]: If not needed, kindly remove.

File: ui/decredmaterial/clickable_list.go [hunk 38]
```diff
@@ -53,31 +66,41 @@ func (cl *ClickableList) handleClickables(count int) {
...
-		}
-		row := cl.clickables[i].Layout(gtx, func(gtx layout.Context) layout.Dimensions {
-			return w(gtx, i)
+	if cl.HideScroll {
+		return cl.List.Layout(gtx, count, func(gtx layout.Context, i int) layout.Dimensions {
+			return cl.row(gtx, count, i, w)
 		})
+	}
+	gtx.Constraints.Min.X = gtx.Constraints.Max.X
+	return material.List(cl.theme.Base, cl.list).Layout(gtx, count, func(gtx C, i int) D {
```
Comment (Sirmorrison) [inline ui/decredmaterial/clickable_list.go]: Why use `gio material` directly? Kindly add this as a file on the `decredmaterial` package. `list.go`.
See here for examples.
https://github.com/planetdecred/godcr/blob/master/ui/decredmaterial/checkbox.go.
https://github.com/planetdecred/godcr/blob/master/ui/decredmaterial/label.go.

File: ui/page/seedbackup/verify_seed.go [hunk 39]
```diff
@@ -198,26 +207,23 @@ func (pg *VerifySeedPage) Layout(gtx C) D {
...
-				func(gtx C) D {
+				}),
+				layout.Rigid(func(gtx C) D {
 					return layout.Inset{
 						Bottom: values.MarginPadding96,
 					}.Layout(gtx, func(gtx C) D {
-						return pg.seedList.Layout(gtx, len(pg.multiSeedList), func(gtx C, index int) D {
-							return pg.seedListRow(gtx, index, pg.multiSeedList[index])
+						gtx.Constraints.Min.X = gtx.Constraints.Max.X
+						return material.List(pg.Theme.Base, pg.list).Layout(gtx, len(pg.multiSeedList), func(gtx C, i int) D {
```
Comment (Sirmorrison) [inline ui/page/seedbackup/verify_seed.go]: Why are we using `material.List`? Kindly add this as a file on the decredmaterial package. list.go.
See here for examples.
https://github.com/planetdecred/godcr/blob/master/ui/decredmaterial/checkbox.go.
https://github.com/planetdecred/godcr/blob/master/ui/decredmaterial/label.go.

File: ui/page/tickets/overview.go [hunk 40]
```diff
@@ -132,20 +140,28 @@ func (pg *Page) loadPageData() {
...
-			},
-		}
-		return pg.ticketPageContainer.Layout(gtx, len(sections), func(gtx C, i int) D {
-			return sections[i](gtx)
+			})
+		},
+	}
+	gtx.Constraints.Min.X = gtx.Constraints.Max.X
+	return layout.Inset{Top: values.MarginPadding24}.Layout(gtx, func(gtx layout.Context) layout.Dimensions {
+		return material.List(pg.Theme.Base, pg.list).Layout(gtx, len(widgets), func(gtx C, i int) D {
```
Comment (Sirmorrison) [inline ui/page/tickets/overview.go]: Why use gio material directly? Kindly add this as a file on the decredmaterial package. list.go.
See here for examples.
https://github.com/planetdecred/godcr/blob/master/ui/decredmaterial/checkbox.go.
https://github.com/planetdecred/godcr/blob/master/ui/decredmaterial/label.go.

**Review by Sirmorrison — DISMISSED** [review]
Scrollbar not visible on the Overview page.
image

No scroll bar on tx details page.
image

No scroll bar on wallet page when all the wallets are expanded.
image

Not so sure why this icon is on the left of your page instead of the right.
image

The scroll padding on the proposals page is not uniform as the against tx and ticket pages.
image

Proposal details content area does not have a scroll bar.
image

Setting page does not have a scroll bar.
image

* Wallet Log Page and statistics page do not also have scroll bars.

The idea of this PR is to add scroll bars to all pages that scrolls. Meaning any page that has the `list` layout.

File: ui/decredmaterial/clickable_list.go (hunk 34 above)
Comment (dreacot) [reply ui/decredmaterial/clickable_list.go]: `isScrollBarHidden` seems like a better variable name

**Review by dreacot — COMMENTED** [review]
The all tickets page shows a long scrollbar, but the remainder of the page is blank
Screenshot 2021-10-25 at 6 21 20 PM

This page should have a scrollbar
Screenshot 2021-10-25 at 6 23 54 PM

This modal should have a scrollbar
Screenshot 2021-10-25 at 6 24 43 PM

This page is scrolled to the bottom but the scrollbar isn't
Screenshot 2021-10-25 at 6 26 37 PM

File: ui/page/transactions_page.go [hunk 41]
```diff
@@ -19,22 +20,25 @@ type TransactionsPage struct {
...
+	transactions    []dcrlibwallet.Transaction
+	wallets         []*dcrlibwallet.Wallet
 }
 
 func NewTransactionsPage(l *load.Load) *TransactionsPage {
 	pg := &TransactionsPage{
-		Load:            l,
-		container:       layout.Flex{Axis: layout.Vertical},
+		Load: l,
+		// container:       layout.Flex{Axis: layout.Vertical},
```
Comment (dreacot) [inline ui/page/transactions_page.go:38]: remove unused code

**Review by dreacot — DISMISSED** [review]
LGTM

**Review by dreacot — APPROVED** [review]
LGTM

**Pull request merged**